### ISSUE TITLE
deploy script does not seem to set correct permissions.

### ISSUE THREAD

**Issue by 2sn** [ADACS-Australia/starfit-server#11]
after fresh setup, the log says `/var/log/httpd/error_log`
```
[Fri Oct 14 07:11:11.810585 2022] [cgid:error] [pid 775:tid 875] [client 130.194.160.128:48318] AH01215: stderr from /var/www/html/index.html: PermissionError: [Errno 13] Permission denied: '/var/www/html/header.py'
```

**Comment by 2sn** [comment]
```
[root@starfit fedora]# ls -latr /var/www/html/header.py
-rw-r-----. 1 root root 142 Oct 12 04:47 /var/www/html/header.py
```

**Comment by conradtchan** [comment]
It looks like this wasn't fully fixed, and is still carrying through the wrong permissions caused by your umask. 

We'll fix it next week, but in the meantime, I'll deploy it from my machine so that it can get going again.

**Comment by 2sn** [comment]
@dliptai @conradtchan I leave this at starfit.xyz
```
[fedora@starfit ~]$ ls -latr /var/www/html/
total 68
drwxr-xr-x. 4 root root 4096 Jun 17 11:13 ..
-rw-r-----. 1 root root 1853 Oct  7 05:28 README.txt
drwxr-x---. 2 root root 4096 Oct  7 05:28 images
-rw-r-----. 1 root root   20 Oct 12 04:47 config.yml
-rw-r-----. 1 root root  967 Oct 12 04:47 scripts.js
-rw-r-----. 1 root root 1685 Oct 12 04:47 index.css
-rw-r-----. 1 root root  142 Oct 12 04:47 header.py
-rwxr-xr-x. 1 root root  266 Oct 12 04:47 unsubscribe
-rw-r-----. 1 root root 6282 Oct 13 06:11 utils.py
drwxr-x---. 2 root root 4096 Oct 13 06:11 templates
-rwxr-xr-x. 1 root root 1766 Oct 13 06:11 run
-rw-r-----. 1 root root   40 Oct 13 06:11 robots.txt
-rw-r-----. 1 root root 5907 Oct 13 06:11 job.py
-rwxr-xr-x. 1 root root 1115 Oct 13 06:11 index.html
drwxr-xr-x. 4 root root 4096 Oct 13 06:11 .
lrwxrwxrwx. 1 root root    9 Oct 14 06:55 data -> /srv/data
```
for debugging, but change on `starfit.org` to have a running server

**Comment by 2sn** [comment]
@conradtchan Thanks.  Server is running!

**Comment by 2sn** [comment]
@conradtchan @dliptai 
I think for the server setup (and ansible script in general) it seems a necessity to to set all permissions of copied files explicitly to ensure stable deployment. (or, maybe defaults can be set in copy/scp process for ansible in general?).

**Comment by dliptai** [comment]
just pushed a (hopeful) fix 0c62370322a4571896a82838b10b38f1d7990d11

**Comment by dliptai** [comment]
more robust fix in 91ed8a9cee077c0a1fdb47ad8b46d527b286a208

**Comment by 2sn** [comment]
I did not see any renewed permission issues when I did the deploy onto `starfit.xyz` last night (except the one issues that was to to the refactored python code to use `pathlib` more than `os.path`.

So I could try a deploy to the main `starfit.org` site with the new script?
